packages/keycloak-user-management/src/components/UserList/TableActions/utils.tsx

```diff
@@ -1,35 +1,132 @@
 import { KeycloakService } from '@opensrp/keycloak-service';
+import { OpenSRPService } from '@opensrp/react-utils';
 import { sendErrorNotification, sendSuccessNotification } from '@opensrp/notifications';
 import { removeKeycloakUsers } from '../../../ducks/user';
-import { KEYCLOAK_URL_USERS } from '../../../constants';
+import {
+  KEYCLOAK_URL_USERS,
+  PRACTITIONER,
+  OPENSRP_CREATE_PRACTITIONER_ENDPOINT,
+  DELETE_PRACTITIONER_ROLE,
+} from '../../../constants';
 import lang, { Lang } from '../../../lang';
+import { Practitioner } from '@opensrp/team-management';
 
 /**
- * Delete keycloak user
+ * Delete keycloak user and practitioner
  *
  * @param {Function} removeKeycloakUsersCreator - remove users action creator
  * @param {string} keycloakBaseURL - keycloak api base URL
+ * @param opensrpBaseURL - opensrp api base url
  * @param {string} userId - id of user to be deleted
  * @param {Function} isLoadingCallback - callback function that sets loading state
  * @param {Lang} langObj - lang
  * @returns {void}
  */
-export const deleteUser = (
+export const deleteUser = async (
   removeKeycloakUsersCreator: typeof removeKeycloakUsers,
   keycloakBaseURL: string,
+  opensrpBaseURL: string,
   userId: string,
   isLoadingCallback: (loading: boolean) => void,
   langObj: Lang = lang
-): void => {
+) => {
+  // start loader
+  isLoadingCallback(true);
+
+  // get tied practitioner from base user Id
+  const practitioner = await getPractitionerByUserId(
+    OPENSRP_CREATE_PRACTITIONER_ENDPOINT,
+    userId,
+    opensrpBaseURL
+  ).catch(() => {
+    sendErrorNotification(langObj.ERROR_OCCURED);
+    // stop loader
+    isLoadingCallback(false);
```

Comment:
we can move this to a `finally` block

Reply:
This stops the loader only when this request fails. If it succeeds it moves to the other requests. If we move it to finally here, the loader would stop prematurely